internal/model/model.go

```diff
@@ -1112,12 +1117,16 @@ func (m *Model) ScanFolders() {
 		go func() {
 			err := m.ScanFolder(folder)
 			if err != nil {
+				errors.Lock()
+				errors.m[folder] = err
+				errors.Unlock()
 				m.cfg.InvalidateFolder(folder, err.Error())
 			}
 			wg.Done()
 		}()
 	}
 	wg.Wait()
+	return errors.m
```

Comment:
This could be replaced with something more elegant:
https://github.com/syncthing/syncthing/blob/master/internal/discover/discover.go#L117

Or at least, store the map and the mut as simple variables, no need to have them in an anonymous struct.